### ISSUE TITLE
Trying to search tweet by tweet ID

### ISSUE THREAD

**Issue by MsVoxxie** [Rishikant181/Rettiwt-API#362]
Is this a current possibility? Or is there some other method? There is no current documentation on TweetFilter that I can find.

**Comment by Rishikant181** [comment]
Look into [tweet.details](https://rishikant181.github.io/Rettiwt-API/classes/TweetService.html#details) method.

Usage example:
```
const { Rettiwt } = require('rettiwt-api');

// Creating a new Rettiwt instance using the API_KEY
const rettiwt = new Rettiwt(API_KEY);

// Fetching the details of the tweet with the given <id>
rettiwt.tweet.details('<id>')
.then(details => {
	...
})
.catch(error => {
	...
});
```

**Comment by MsVoxxie** [comment]
> Look into [tweet.details](https://rishikant181.github.io/Rettiwt-API/classes/TweetService.html#details) method.
> 
> Usage example:
> 
> ```
> const { Rettiwt } = require('rettiwt-api');
> 
> // Creating a new Rettiwt instance using the API_KEY
> const rettiwt = new Rettiwt(API_KEY);
> 
> // Fetching the details of the user whose username is <username>
> rettiwt.tweet.details('<id>')
> .then(details => {
> 	...
> })
> .catch(error => {
> 	...
> });
> ```

Oh I completely missed this, I apologise! Thank you so much

**Comment by MsVoxxie** [comment]
I apologise again, one second question. Is it a limitation of the method which this package is using to obtain the video url of a tweet? Or is it possible to get the video / media of a tweet?

**Comment by Rishikant181** [comment]
The video/media URL can be obtained from the [entities.media](https://rishikant181.github.io/Rettiwt-API/classes/TweetEntities.html#media) field of the returned [tweet](https://rishikant181.github.io/Rettiwt-API/classes/Tweet.html). The media can then be downloaded using a separate HTTP request to the received URL. 

Since this API uses reverse engineered API endpoints of the Twitter client applications, this is the way by which media contents are received.

TL;DR The returned tweet contains the URL to the media and not the full media file.

**Comment by MsVoxxie** [comment]
> The video/media URL can be obtained from the [entities.media](https://rishikant181.github.io/Rettiwt-API/classes/TweetEntities.html#media) field of the returned [tweet](https://rishikant181.github.io/Rettiwt-API/classes/Tweet.html). The media can then be downloaded using a separate HTTP request to the received URL.
> 
> Since this API uses reverse engineered API endpoints of the Twitter client applications, this is the way by which media contents are received.
> 
> TL;DR The returned tweet contains the URL to the media and not the full media file.

I see, Though when testing tweet ID's of tweets with videos the media tab returns jpg's or png's and never a video to be requested

**Comment by Rishikant181** [comment]
Yes, that's because after receiving the Tweet data, the Twitter web app proceeds to fetch the actual file specified by the URL and then renders it in the display via a separate HTTP request.

**Comment by Rishikant181** [comment]
Sorry I understood your question wrong :P. 

And 

> Is it a limitation of the method which this package is using to obtain the video url of a tweet? 

Yes, it's a limitation as of now (probably because dumb me just totally forgot about videos inside tweets).

I'm opening up a separate issue and will try to implement it in the next feature update. Thanks for the feedback!

**Comment by MsVoxxie** [comment]
Oh awesome, I'd absolutely love video endpoints being reachable. Thanks so much :)
